### PR TITLE
Extracting CheckoutRepository step class from CloneRepository step class towards fixing the bug related to issue #65.

### PR DESCRIPTION
**Pull request by fermadeiral**

### CODE REVIEW

File: repairnator/src/main/java/fr/inria/spirals/repairnator/process/git/GitHelper.java [hunk 1]
```diff
@@ -11,6 +11,7 @@
 import okhttp3.Response;
 import org.eclipse.jgit.api.Git;
 import org.eclipse.jgit.api.MergeCommand;
+import org.eclipse.jgit.api.MergeResult;
```
Comment (surli) [inline repairnator/src/main/java/fr/inria/spirals/repairnator/process/git/GitHelper.java]: do you need it?

File: repairnator/src/main/java/fr/inria/spirals/repairnator/process/step/CheckoutRepository.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Created by fernanda on 02/03/17.
+ */
+public class CheckoutRepository extends AbstractStep {
```
Comment (surli) [inline repairnator/src/main/java/fr/inria/spirals/repairnator/process/step/CheckoutRepository.java]: ok so this one should be abstract then: you don't want to execute the step as it.

File: repairnator/src/main/java/fr/inria/spirals/repairnator/serializer/AbstractDataSerializer.java [hunk 3]
```diff
@@ -60,6 +60,12 @@ protected String getPrettyPrintState(ProjectState state, GatherTestInformation g
             case PREVIOUSVERSIONISNOTINTERESTING:
                 return "previous build is not interesting";
 
+            case BUILDCHECKEDOUT:
```
Comment (surli) [inline repairnator/src/main/java/fr/inria/spirals/repairnator/serializer/AbstractDataSerializer.java]: I'm not sure this is the only place you have to add new cases... did you check on google sheets serializer?

File: repairnator/src/main/java/fr/inria/spirals/repairnator/process/git/GitHelper.java (hunk 1 above)
Comment (fermadeiral) [inline repairnator/src/main/java/fr/inria/spirals/repairnator/process/git/GitHelper.java]: It was not on purpose. I was trying to do something after merge but then I gave up for now and I forgot to remove that import.

File: repairnator/src/main/java/fr/inria/spirals/repairnator/process/step/CheckoutRepository.java (hunk 2 above)
Comment (fermadeiral) [inline repairnator/src/main/java/fr/inria/spirals/repairnator/process/step/CheckoutRepository.java]: Ok.

File: repairnator/src/main/java/fr/inria/spirals/repairnator/serializer/AbstractDataSerializer.java (hunk 3 above)
Comment (fermadeiral) [inline repairnator/src/main/java/fr/inria/spirals/repairnator/serializer/AbstractDataSerializer.java]: Yes, this is not the only place.

**Pull request merged**